### ISSUE TITLE
Browser build : async / await references in the distributed code

### ISSUE THREAD

**Issue by Donov4n** [crossbario/autobahn-js#518]
Hello and thanks for this super useful project !

I don't know if it's the right place to open an issue about the browser build repository (`crossbario/autobahn-js-browser`) but since issues aren't enabled on the project, I figured it had to be here.

--

The browser build embed number of polyfills, but it seems that it also have [references to async / await](https://github.com/crossbario/autobahn-js-browser/blob/master/autobahn/autobahn.js#L5509) that are not polyfilled and the browser support for this feature is [relatively new and does not include IE](https://caniuse.com/#feat=async-functions) at all.  

I doubt it was done on purpose (= the async / await in the distributed build), especially when we consider that the code contains [polyfill for IE 9](https://github.com/crossbario/autobahn-js-browser/blob/master/autobahn/autobahn.js#L268).

But if that's the case, if it's meant to be, this should probably have been included in a major version (because it's a breaking change) and not a minor version as it was the case (20.2.2 => 20.4.1) (at least if the project follow semver).

---

__EDIT:__ This seems to concerns the unminified version only, the minified version does not seem to contain any reference to async / await.

**Comment by om26er** [comment]
> I doubt it was done on purpose (= the async / await in the distributed build), especially when we consider that the code contains [polyfill for IE 9](https://github.com/crossbario/autobahn-js-browser/blob/master/autobahn/autobahn.js#L268).

We made the change intentionally, see https://forum.crossbar.io/t/whats-the-minimal-version-of-ecmascript-supported/1588 -- We kind of consider IE9 "ancient" software.

> But if that's the case, if it's meant to be, this should probably have been included in a major version (because it's a breaking change) and not a minor version as it was the case (20.2.2 => 20.4.1) (at least if the project follow semver).

Actually, we don't follow semver ;-) our versioning is inspired by Twisted's release cycle. YY.MM.mm, where YY is the last two digits of the year of the release, MM is the number of the release in the year, and mm is the number of the patch release.

**Comment by Donov4n** [comment]
> We made the change intentionally, see https://forum.crossbar.io/t/whats-the-minimal-version-of-ecmascript-supported/1588

The linked forum post does not seem to be specifically related to the browser build (`crossbario/autobahn-js-browser`) but rather to `crossbario/autobahn-js`.  

For the browser distribution,  there is a kind of transpilation + polyfill, etc. except for this, that's what struck me as odd.
But okay ... if it's intentional ... :) 

Maybe the README should contains a compatibility paragraph, since the lib doesn't support IE at all (including IE 11) and only recent browser version (?)

For the SemVer part, good to know (and redhibitory to me, if each dependency follow a specific convention for the versionning, and we have to hold them all back, it really gets too complicated)

I'm closing, thank for the answer.

**Comment by Donov4n** [comment]
Note: If this project use ES+ features like async / await without transpilling, the `engines.node` in `package.json` should probably be updated according to the version of node that supports these features.

As the time i'm writing, the package.json contain [4.6.2](https://github.com/crossbario/autobahn-js/blob/master/package.json#L6), but Node support Async / Await since Node ... [7.10.1](https://node.green/#ES2017-features-async-functions).

**Comment by om26er** [comment]
> As the time i'm writing, the package.json contain [4.2.6](https://github.com/crossbario/autobahn-js/blob/master/package.json#L6), but Node support Async / Await since Node ... [7.10.1](https://node.green/#ES2017-features-async-functions).

You are right https://node.green/, I will take care of that

**Comment by oberstet** [comment]
yeah, AutobahnJS is using calver, not semver. but yes, we probably should:

* make node >=7.10.1. required (for node)
* mention the requirement of async/await in https://github.com/crossbario/autobahn-js/blob/master/doc/changelog.md
